crates/tasks/src/lib.rs

```diff
@@ -304,6 +308,25 @@ pub struct TaskExecutor {
 // === impl TaskExecutor ===
 
 impl TaskExecutor {
+    /// Returns the current `TaskExecutor` if one has been initialized
+    pub fn current() -> Option<Self> {
+        GLOBAL_EXECUTOR.get().cloned()
+    }
+    /// Returns the current `TaskExecutor` or creates one if none exists
+    pub fn current_or_create() -> Self {
+        Self::current().unwrap_or_else(|| {
+            let handle = Handle::try_current().unwrap_or_else(|_| {
+                static RT: OnceLock<Runtime> = OnceLock::new();
+                let rt = RT.get_or_init(|| {
+                    tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap()
```

Comment:
since this type is used to spawn new tasks we should use the default builder and not current_thread